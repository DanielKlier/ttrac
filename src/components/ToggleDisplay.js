import * as React from 'react';
import PropTypes from 'prop-types';

const shouldHide = props => {
    if (props.show !== null) {
        return !props.show;
    }
    else if (props.hide !== null) {
        return !!props.hide;
    }
    else {
        return false;
    }
};

function toggleDisplay(props) {

    if (props.if === false) {
        return null;
    }

    const newProps = Object.assign({}, props);
    propsToRemove.forEach(p => {
        if (p in newProps) {
            delete newProps[p];
        }
    });

    const className = `${shouldHide(props) ? 'hidden' : ''}`;

    return (
        <span className={className}>{props.children}</span>
    );
}

toggleDisplay.PropTypes = {
    if: PropTypes.bool,
    show: PropTypes.bool,
    hide: PropTypes.bool
};

const propsToRemove = Object.keys(toggleDisplay.PropTypes);

export default toggleDisplay;
