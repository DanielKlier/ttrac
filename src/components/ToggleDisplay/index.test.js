import React from 'react';
import ToggleDisplay from './index';
import renderer from 'react-test-renderer';

test('it returns null when if prop is falsy', () => {
    const component = renderer.create(
        <ToggleDisplay if={false}>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toBeNull();
});

test('it renders the component when if prop is truthy or not present', () => {
    let component = renderer.create(
        <ToggleDisplay>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <ToggleDisplay if={true}>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('it handles the show prop correctly', () => {
    let component = renderer.create(
        <ToggleDisplay show={true}>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <ToggleDisplay show={false}>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('it handles the hide prop correctly', () => {
    let component = renderer.create(
        <ToggleDisplay hide={true}>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <ToggleDisplay hide={false}>
            <div>Hello</div>
        </ToggleDisplay>
    );

    expect(component.toJSON()).toMatchSnapshot();
});
