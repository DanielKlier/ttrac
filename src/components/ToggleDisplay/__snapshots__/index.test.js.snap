// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it handles the hide prop correctly 1`] = `
<span
  className="hidden"
>
  <div>
    Hello
  </div>
</span>
`;

exports[`it handles the hide prop correctly 2`] = `
<span
  className=""
>
  <div>
    Hello
  </div>
</span>
`;

exports[`it handles the show prop correctly 1`] = `
<span
  className=""
>
  <div>
    Hello
  </div>
</span>
`;

exports[`it handles the show prop correctly 2`] = `
<span
  className="hidden"
>
  <div>
    Hello
  </div>
</span>
`;

exports[`it renders the component when if prop is truthy or not present 1`] = `
<span
  className=""
>
  <div>
    Hello
  </div>
</span>
`;

exports[`it renders the component when if prop is truthy or not present 2`] = `
<span
  className=""
>
  <div>
    Hello
  </div>
</span>
`;
